src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java

```diff
@@ -0,0 +1,310 @@
+			returnSb.append("Info Items:").append(NEWLINE);
+			for (int idx = 0; idx == infoList.size() - 1; idx++) {
+				returnSb.append(infoList.get(idx).show()).append(NEWLINE);
+				// Ex: 1:31: A draft of the next version of CSP deprecates
+				// report-uri in favour of a new report-to directive.
+			}
+		}
+		return returnSb.toString();
+	}
+
+	/**
+	 * Extracts a list of headers, and returns them without changing
+	 * their cases.
+	 * @param msg HTTP Response message
+	 * @param header The header field(s) to be found
+	 * @return list of the matched headers
+	 */
+	private List<String> getHeaderField(HttpMessage msg, String header) {
+		List<String> matchedHeaders = new ArrayList<>();
+		String headers = msg.getResponseHeader().toString();
+		String[] headerElements = headers.split("\\r\\n");
+		Pattern pattern = Pattern.compile("^"+header, Pattern.CASE_INSENSITIVE);
+		for (String hdr : headerElements) {
+			Matcher matcher = pattern.matcher(hdr);
+			if (matcher.find()) {
+				String match = matcher.group();
+				matchedHeaders.add(match);
+			}
+		}
+		return matchedHeaders;
+	}
+	
+	private List<String> getAllowedWildcardSources(String policyText, Origin origin) {
+		
+		List<String> allowedSources = new ArrayList<String>();
+		Policy pol = ParserWithLocation.parse(policyText, origin);
+
+		if (pol.allowsScriptFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("script-src");
+		}
+		if (pol.allowsStyleFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("style-src");
+		}
+		if (pol.allowsImgFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("img-src");
+		}
+		if (pol.allowsConnectTo(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("connect-src");
+		}
+		if (pol.allowsFrameFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("frame-src");
+		}
+		if (pol.allowsFrameAncestor(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("frame-ancestor");
+		}
+		if (pol.allowsFontFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("font-src");
+		}
+		if (pol.allowsMediaFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("media-src");
+		}
+		if (pol.allowsObjectFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("object-src");
+		}
+		if (pol.allowsManifestFromSource(URI.parse(WILDCARD_URI))) {
+			allowedSources.add("manifest-src");
+		}
+		return allowedSources;
+	}
+	
+	@Override
+	public int getPluginId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	private String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	private String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	private void raiseAlert(HttpMessage msg, String name, int id, String description, String param, int risk) {
+		String ALERT_NAME_PREFIX = getName() + ": ";
+		name = StringUtils.isEmpty(name) ? getName() : name;

```

Comment:
These two statements could be replaced with:
```Java
String alertName = StringUtils.isEmpty(name) ? getName() : getName() + ": " + name;
```
otherwise we might end up with "CSP Scanner: CSP Scanner" (if in future versions the `name` is empty).